### PR TITLE
Create change_pivot.py #429

### PR DESCRIPTION
**Pull request by og76**

### CODE REVIEW

**Comment by og76** [comment]
for #429

File: nodes/matrix/change_pivot.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import bpy
+from bpy.props import *
+from ... tree_info import keepNodeState
+from ... events import executionCodeChanged
+from ... base_types.node import AnimationNode
+
+inputTypeItems = [
```
Comment (JacquesLucke) [inline nodes/matrix/change_pivot.py]: would be nice to have all 5 values here for each item. Most importantly the last number which will be used as identifier by Blender. In my experience this is more 'stable' when we extend nodes later.

File: nodes/matrix/change_pivot.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    @keepNodeState
+    def generateSockets(self):
+        self.inputs.clear()
+        self.outputs.clear()
+        
+        type = self.inputType
+        condition = "List" if "LIST" in type else ""
+        name = names[type]
+        Name = name.capitalize() + " " + condition
```
Comment (JacquesLucke) [inline nodes/matrix/change_pivot.py]: maybe rename 'Name' to 'displayName' or so

**Comment by JacquesLucke** [comment]
To be hones, I don't understand how the getExecutionCode function works atm but I'll trust you here that it works. I'll play with it later a bit and maybe clean it up a bit so that I can understand it as well :D

File: nodes/matrix/change_pivot.py (hunk 2 above)
Comment (og76) [inline nodes/matrix/change_pivot.py]: it only make diff between the item/list case. like:

 - matrix = **stuff**
 - matrixList = [ **stuff** for x in matrixList]

so I format the  _" = "_ vs _"List = "_  and at the end _""_(nothing)  vs  _" for x in matrixList]"_

**Comment by JacquesLucke** [comment]
I'm still not sure if "Pivot Matrix" is the best name here but I don't know any better so it is ok.
Maybe we should give the node a more specific name like "Change Matrix Pivot" or "Change Transformation Pivot" because just "Change Pivot" is too unspecific.

**Comment by og76** [comment]
Should i also include it as one of the polygon cases? In the transform poly? That would break the node (version)

Names?
 maybe it should go like convert rotations, having different Different names/search for each case:
 "Change Matrix Pivot",  "Change Rotation Pivot",  "Change Matrix List Pivot" etc

 as it kinda goes along with transform nodes? Problem is that I included more than matrix transform ..

Not sure, maybe I should also have some search tags to include "parent matrix" or "reference" something.

**Comment by JacquesLucke** [comment]
I'm testing the node right now and I don't understand everything.
I mean changing the pivot for a matrix makes sense but no for the rest... Also "Pivot for Vector" doesn't change the vector at all

**Comment by JacquesLucke** [comment]
are the change rotation/vector pivot modes really needed because the seam to be a bit confusion to me

**Comment by JacquesLucke** [comment]
Oh and it is very uneffective to calculate the inverted pivot matrix for each matrix in a list. It's better to calculate it beforehand

**Pull request closed without merging**